cocotb/scheduler.py

```diff
@@ -585,27 +597,20 @@ def add(self, coroutine):
         useful error messages in the event of common gotchas.
         """
         if isinstance(coroutine, cocotb.decorators.coroutine):
-            self.log.critical(
-                "Attempt to schedule a coroutine that hasn't started")
-            coroutine.log.error("This is the failing coroutine")
-            self.log.warning(
-                "Did you forget to add parentheses to the @test decorator?")
-            self._test_result = TestError(
-                "Attempt to schedule a coroutine that hasn't started")
-            self._terminate = True
-            return
+            raise TypeError(
+                "Attempt to schedule a coroutine that hasn't started: {}.\n"
+                "Did you forget to add parentheses to the @test decorator?"
+                .format(coroutine)
+            )
 
         elif not isinstance(coroutine, cocotb.decorators.RunningCoroutine):
-            self.log.critical(
+            raise TypeError(
                 "Attempt to add something to the scheduler which isn't a "
-                "coroutine")
-            self.log.warning(
-                "Got: %s (%s)" % (str(type(coroutine)), repr(coroutine)))
-            self.log.warning("Did you use the @coroutine decorator?")
-            self._test_result = TestError(
-                "Attempt to schedule a coroutine that hasn't started")
-            self._terminate = True
-            return
+                "coroutine.\n"
+                "Got: {} ({!r})\n"
+                "Did you use the @coroutine decorator?"
```

Comment:
`@coroutine` -> `@cocotb.coroutine` (just to be helpful to users, this is the form we typically document and use it)

Reply:
Sure